modules: CMakeLists: Update LVGL to 9.4.0

faxe1008 opened the pull request:
Update LVGL CMakeLists for latest release.

faxe1008 committed:
manifest: Update LVGL to v9.4

Update the west yaml to point to the new LVGL version.
Update CMakeLists and samples accordingly.

Signed-off-by: Fabian Blatz <fabianblatz@gmail.com>

faxe1008 commented:
v9.4.0 is officially tagged, updated the module PR accordingly.

uLipe reviewed (dismissed):
Great the VGLite kernel drivers were added as well, thank you @faxe1008 !

faxe1008 committed:
tests: lib: gui: lvgl: Add missing DYNAMIC_THREAD dependency

This patch adds a CONFIG_THREAD_STACK_INFO to properly enable
CONFIG_DYNAMIC_THREAD.

Signed-off-by: Fabian Blatz <fabianblatz@gmail.com>

faxe1008 committed:
modules: lvgl: Fix thread deletion in OSAL

Do not delete the thread from within the thread callback function,
secure thread_delete against trying to delete a terminated thread.

Signed-off-by: Fabian Blatz <fabianblatz@gmail.com>

faxe1008 commented:
@uLipe found that the osal was now missing `lv_sleep_ms` which led me to fixing the test depedencies and the thread deletion logic.

VynDragon reviewed (dismissed):
Ran my usages, and built and ran samples, and LGTM.

pdgendt reviewed (changes requested):
The manifest file change and related changes to `modules/lvgl/CMakeLists.txt` need to be done in a single commit, otherwise the git history isn't bisectable because the version in between does not compile.

Not sure about the other two fix commits as well.

Rendering looks good!

faxe1008 commented:
Hmm not sure, actually. I mean it makes sense, for bisectability, but I separated the commits in the past as well. Do you combine them for nanopb as well?

JarmouniA reviewed (dismissed):
Tested on STM32H750B-DK

uLipe reviewed (dismissed):
Sorry for taking too long time to review this (I was no holidays). 

Tests with my IMXRT1176-EVKB is okay as well.
